File: services/vets-service/src/test/java/com/github/antrakos/petclinic/vets/IntegrationTest.java
package com.github.antrakos.petclinic.vets;

import com.github.antrakos.petclinic.vets.config.TestContainersConfig;
import org.junit.jupiter.api.TestInstance;
import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.context.annotation.Import;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@SpringBootTest
@AutoConfigureMockMvc
@TestInstance(TestInstance.Lifecycle.PER_CLASS)
@Import(TestContainersConfig.class)
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
public @interface IntegrationTest {
}

File: services/visits-service/src/test/kotlin/com/github/antrakos/petclinic/visits/IntegrationTest.kt
package com.github.antrakos.petclinic.visits

import com.github.antrakos.petclinic.visits.config.TestContainersConfig
import org.junit.jupiter.api.TestInstance
import org.springframework.boot.test.autoconfigure.web.reactive.AutoConfigureWebTestClient
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.context.annotation.Import
import org.springframework.test.context.TestConstructor

@SpringBootTest
@AutoConfigureWebTestClient
@Import(TestContainersConfig::class)
@TestInstance(TestInstance.Lifecycle.PER_CLASS)
@TestConstructor(autowireMode = TestConstructor.AutowireMode.ALL)
annotation class IntegrationTest
